nexus/db-queries/src/db/collection_attach.rs

```diff
@@ -859,7 +857,7 @@ mod test {
             dev::test_setup_log("test_attach_missing_collection_fails");
         let mut db = test_setup_database(&logctx.log).await;
         let cfg = db::Config { url: db.pg_config().clone() };
-        let pool = db::Pool::new(&logctx.log, &cfg);
+        let pool = db::Pool::new_qorb_single_host(&cfg);
```

Comment:
There are two flavors of `db::Pool` that are exposed now:

1. One that uses DNS to resolve backends (this will be used in prod, and recommended), and
2. One that uses a hard-coded address for resolution and skips DNS (this is used by a lot of our tests)

I'm using the function name `new_qorb_single_host` to identify case (2) explicitly.